lib/membrane_matroska/parser/matroska.ex

```diff
@@ -0,0 +1,174 @@
+      # 14	both eyes laced in one Block (right eye is first)
+      _other -> :unsupported
+    end
+  end
+
+  @spec parse_doc_type(binary) :: binary
+  def parse_doc_type(bytes) do
+    text = EBML.parse_string(bytes)
+
+    case text do
+      "webm" -> "webm"
+      "matroska" -> "matroska"
+      type -> raise "The file DocType is '#{type}' but it MUST be 'matroska' or 'webm' "
+    end
+  end
+
+  @spec parse_codec_id(binary) :: :opus | :vorbis | :vp8 | :vp9
+  def parse_codec_id(bytes) do
+    text = EBML.parse_string(bytes)
+
+    case text do
+      "A_OPUS" -> :opus
+      "A_VORBIS" -> :vorbis
+      "V_VP8" -> :vp8
+      "V_VP9" -> :vp9
+      "V_MPEG4/ISO/AVC" -> :h264
+      codec -> raise "Matroska contains illegal codec #{codec}"
+    end
+  end
+
+  @spec parse_block(binary) :: %{
+          data: binary,
+          header_flags: %{
+            discardable: boolean,
+            invisible: boolean,
+            lacing: :EBML_lacing | :Xiph_lacing | :fixed_size_lacing | :no_lacing
+          },
+          timecode: integer,
+          track_number: non_neg_integer
+        }
+  def parse_block(bytes) do
+    # track_number is a vint with size 1 or 2 bytes
+    {:ok, {track_number, body}} = EBML.decode_vint(bytes)
+
+    <<timecode::integer-signed-big-size(16), _reserved::3, invisible::1, lacing::2,
+      discardable::1, _unused::1, data::binary>> = body
+
+    %{
+      track_number: track_number,
+      timecode: timecode,
+      header_flags: %{
+        invisible: invisible == 1,
+        lacing: check_lacing_mode(lacing),
+        discardable: discardable == 1
+      },
+      data: data
+    }
+  end
+
+  # https://tools.ietf.org/id/draft-lhomme-cellar-matroska-04.html#rfc.section.6.2.4.4
+  @spec parse_simple_block(binary) :: %{
+          data: binary,
+          header_flags: %{
+            discardable: boolean,
+            invisible: boolean,
+            keyframe: boolean,
+            lacing: :EBML_lacing | :Xiph_lacing | :fixed_size_lacing | :no_lacing
+          },
+          timecode: integer,
+          track_number: non_neg_integer
+        }
+  def parse_simple_block(bytes) do
+    # track_number is a vint with size 1 or 2 bytes
+    {:ok, {track_number, body}} = EBML.decode_vint(bytes)
+
+    <<timecode::integer-signed-big-size(16), keyframe::1, _reserved::3, invisible::1, lacing::2,
+      discardable::1, data::binary>> = body
+
+    %{
+      track_number: track_number,
+      timecode: timecode,
+      header_flags: %{
+        keyframe: keyframe == 1,
+        invisible: invisible == 1,
+        lacing: check_lacing_mode(lacing),
+        discardable: discardable == 1
+      },
+      data: data
+    }
+  end
+
+  defp check_lacing_mode(mode) do
+    case mode do
+      0b00 ->
+        :no_lacing
+
+      _any_lacing ->
+        raise "Demuxing matroska files with laced data is currently not supported"
+        # 0b01 -> :Xiph_lacing
+        # 0b11 -> :EBML_lacing
+        # 0b10 -> :fixed_size_lacing
```

Comment:
I think it's ok to leave it with a comment about what it is. And above the raise, definitely not as part of this clause